core_lib/src/util/pointerevent.cpp

```diff
@@ -0,0 +1,244 @@
+#include "pointerevent.h"
+
+PointerEvent::PointerEvent(QMouseEvent* event)
+{
+    this->mouseEvent = event;
+}
+
+PointerEvent::PointerEvent(QTabletEvent* event, QPointF widgetPos)
+{
+    this->tabletEvent = event;
+    mWidgetPos = this->widgetPos();
+}
+
+PointerEvent::~PointerEvent()
+{
+}
+
+QPoint PointerEvent::pos() const
+{
+    if (this->mouseEvent)
+    {
+        return mouseEvent->pos();
+    }
+    else if (this->tabletEvent)
+    {
+        return this->tabletEvent->pos();
+    }
+    else
+    {
+        // if we land here... the incoming input was
+        // neither tablet nor mouse
+        return mWidgetPos.toPoint();
```

Comment:
Any possibilities the event is not mouse nor tablet?

Reply:
Hmm no I guess not, well not until we implement another event like touch event.